crates/telio-sockets/src/protector/apple.rs

```diff
@@ -291,20 +328,44 @@ impl Sockets {
 }
 
 #[allow(unreachable_code)]
-fn spawn_monitor(sockets: Arc<Mutex<Sockets>>) -> JoinHandle<io::Result<()>> {
+fn spawn_monitors(
+    sockets: Arc<Mutex<Sockets>>,
+) -> (JoinHandle<io::Result<()>>, JoinHandle<io::Result<()>>) {
     spawn_dynamic_store_loop(Arc::downgrade(&sockets));
-    tokio::spawn(async move {
-        loop {
-            let update = {
-                let sockets = sockets.lock();
-                sockets.notify.clone()
-            };
-
-            update.notified().await;
-            let mut sockets = sockets.lock();
-            sockets.rebind_all(true);
+    let nw_path_monitor_monitor_handle = tokio::spawn({
+        let sockets = sockets.clone();
+        let mut notify = PATH_CHANGE_BROADCAST.subscribe();
+        async move {
+            loop {
+                match notify.recv().await {
+                    Ok(()) => {
+                        let mut sockets = sockets.lock();
+                        sockets.rebind_all(false);
```

Comment:
What is the reason of `false` flag in comparison to `true` in the code below?

Reply:
If the interface didn't change, there is no reason to rebind.

Reply:
should we then even call the `rebind_all()`? I guess the function does more stuff than just rebinding then?

Reply:
> should we then even call the `rebind_all()`? I guess the function does more stuff than just rebinding then?

This will call `rebind_all` when some change was reported by the `NWPathMonitor`. In the rebind_all we 'compute' the actually interface that we would use for the current list of interfaces reported (in OS preference order) by `NWPathMonitor` and if the computed interface is different than the previous one, then we do the actual rebinding.